opentech/apply/stream_forms/fields.py

```diff
@@ -1,25 +1,64 @@
-from django.forms import FileInput, FileField
+from django.forms import ClearableFileInput, FileField, CheckboxInput
 
 
-class MultiFileInput(FileInput):
+class MultiFileInput(ClearableFileInput):
     """
     File Input only returns one file from its clean method.
 
     This passes all files through the clean method and means we have a list of
     files available for post processing
     """
-    def __init__(self, *args, attrs={}, **kwargs):
-        attrs['multiple'] = True
-        super().__init__(*args, attrs=attrs, **kwargs)
+    template_name = 'stream_forms/fields/multi_file_field.html'
+
+    input_text = ''
+
+    def __init__(self, *args, **kwargs):
+        self.multiple = kwargs.pop('multiple', True)
+        super().__init__(*args, **kwargs)
+
+    def is_initial(self, value):
+        is_initial = super().is_initial
+        return all(
+            is_initial(file) for file in value
+        )
+
+    def render(self, name, value, attrs=None):
+        if self.multiple:
+            attrs['multiple'] = 'multiple'
+
+        return super().render(name, value, attrs)
 
     def value_from_datadict(self, data, files, name):
-        return files.getlist(name)
+        if hasattr(files, 'getlist'):
+            upload = files.getlist(name)
+        else:
+            upload = files.get(name)
+            if not isinstance(upload, list):
+                upload = [upload]
+
+        checkbox_name = self.clear_checkbox_name(name)
+        checkboxes = {k for k in data if checkbox_name in k}
+        cleared = {
+            i for i, checkbox in enumerate(checkboxes)
```

Comment:
"i" will always start at 0 regardless of the count of the checkbox checked.

What ever checkbox you check the first firewall get deleted. If you check any two checkboxes the first two files will get deleted.

Reply:
You've rebased over my fixes... Will force push over your changes